Kernel/Ordering.cpp

```diff
@@ -289,6 +272,35 @@ Ordering::Result PrecedenceOrdering::compare(Literal* l1, Literal* l2) const
   return comparePredicates(l1, l2);
 } // PrecedenceOrdering::compare()
 
+bool Ordering::containsVar(AppliedTerm s, TermList var)
+{
+  ASS(var.isVar());
+  if (!s.aboveVar) {
+    return s.term.containsSubterm(var);
+  }
+  if (s.term.isVar()) {
+    return (*s.applicator)(s.term.var()).containsSubterm(var);
+  }
+  VariableIterator vit(s.term.term());
+  while (vit.hasNext()) {
+    auto v = vit.next();
+    if ((*s.applicator)(v.var()).containsSubterm(var)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+Ordering::Result Ordering::compare(AppliedTerm lhs, AppliedTerm rhs) const
+{
+  NOT_IMPLEMENTED;
+}
+
+bool Ordering::isGreater(AppliedTerm lhs, AppliedTerm rhs) const
+{
+  NOT_IMPLEMENTED;
+}
```

Comment:
I don't like the pattern of adding virtual functions and then adding throwing errors when functions are not overridden, as this means that people who use classes can't rely on the typesystem and subtyping but need to know which instance of an abstract class is being used where and in which ways this instance is being used. 

I'd suggest instead having 
```
class AppliedTermOrdering : public Ordering {
  Result compare(AppliedTerm lhs, AppliedTerm rhs) const = 0;
  ...
};
```

And implement this interface for `KBO`, but not for `LPO`.
What do you think? @MichaelRawson @mezpusz

Reply:
Fine by me - I also don't like `NOT_IMPLEMENTED` "abstract" methods, but it seems to be prevalent so I was willing to let this slide. Since LPO will probably have this in the future if it doesn't already (I got a bit confused with what @mezpusz does here and what he plans to do) we could also have this be an abstract method in `Ordering`, and have LPO `NOT_IMPLEMENT` it for now.

Reply:
Actually I just saw that LPO has a (sort-of) default-implementation. So I think the function could just be made purely virtual and we could leave away the `NOT_IMPLEMENTED`.

Reply:
Pure-virtual in `Ordering`, `NOT_IMPLEMENTED` in the to-be-axed SKIKBO?